text/1173792/orientation_syrotiuk_pachl.tex

```diff
@@ -0,0 +1,93 @@
+\usepackage[utf8]{inputenc}
+\usepackage[margin=2cm]{geometry}
+\usepackage{enumitem}
+\usepackage{amsmath}
+\usepackage{amsthm}
+\usepackage{mathtools}
+\newtheorem{lemma}{Lemma}
+\newtheorem{theorem}{Theorem}
+\usepackage{amssymb}
+\usepackage[english]{babel}
+\usepackage{csquotes}
+\usepackage{graphicx}
+
+
+\usepackage[colorlinks=false]{hyperref}
+\usepackage[
+backend=biber,
+style=alphabetic,
+sorting=ynt
+]{biblatex}
+\addbibresource{biblio.bib}
+\usepackage{titlesec}
+\usepackage{tocloft}
+
+
+\begin{document}
+\section{Ring orientation algorithm}
+The goal of \cite{10.1007/BFb0019813} is to find a consistent orientation in a ring of size $N$ of anonymous processes, which means they don't have any type of identifiers or access to randomness. Without these constraints there exist $O(N\log N)$ ring orientation distributed algorithms. For example, if processes have unique identifiers we can elect a leader and then have the leader decide the orientation.
+
+In our algorithm processes will only know the size of the ring. The ring is anonymous (processes are indistinguishable) and asynchronous. Each process has two channels: left and right. Presented algorithm sends $O(N^{3/2})$ messages on average, under the assumption that local orientations for processes are chosen independently.
+
+\section{The algorithm}
+The algorithm terminates with the indication, in at least process, that the orientation in this process is shared by majority of processes. The other option is that there is no majority (the numbers of processes in both orientations are equal) and then, in our model, we cannot decide a consistent ring orientation.
+
+Message has two fields: $vote$ and $distance$. Furthermore, when we receive a message we know from which neighbour (left, right) it came.
+
+Every process executes the following algorithm:
+\begin{enumerate}
+    \item Send message with $vote = 1, distance = 1$ to right neighbour. 
+    \item Repeat until DONE:
+    \begin{enumerate}
+        \item receive $vote, distance$ from neighbour $from$
+        \item if $distance = N$ go to step 3 (DONE)
+        \item if $from = left$ send $vote+1, distance+1$ to right neighbour
+        \item else if $from = right$ and $vote > 0$ send $vote-1, distance+1$ to left neighbour.
+    \end{enumerate}
+    \item DONE: 
+    \begin{enumerate}
+        \item If $vote > 0$ then our orientation agrees with orientations of majority of processes.
+        \item Else if $vote = 0$ there is no majority.
+    \end{enumerate}
+\end{enumerate}
+
+\section{Correctness}
+Proof of correctness is based on combinatorial results about sequences of zeros and ones. One represent first type of orientation, zero the opposite orientation.
+
+In a sequence $s$ of zeros and ones we define $\#_0(s)$ and $\#_1(s)$ as number of zeros and ones in $s$, respectively. We call $s$ \textit{dominating} if $\#_1(s) > \#_0(s)$. We call $s$ \textit{weakly dominating} if $\#_1(s) \geq \#_0(s)$.
+
+\begin{lemma}

```

Comment:
I don't get it: for s = 110 we have p = 2, q = 1, but according to definition all cyclic shifts (110, 101, 011) are dominating. Either the definition or its usage is wrong.

Reply:
Now the definition is fixed